client/my-sites/stats/activity-log/index.jsx

```diff
@@ -317,21 +321,48 @@ class ActivityLog extends Component {
 			m.subtract( 1, 'day' )
 		) {
 			const dayEnd = m.endOf( 'day' ).valueOf();
-			activityDays.push(
-				<ActivityLogDay
-					applySiteOffset={ this.applySiteOffset }
-					requestedRestoreActivityId={ requestedRestoreActivityId }
-					rewindConfirmDialog={ rewindConfirmDialog }
-					disableRestore={ disableRestore }
-					hideRestore={ ! rewindEnabledByConfig || ! isPressable }
-					isRewindActive={ isRewindActive }
-					key={ dayEnd }
-					logs={ get( logsGroupedByDay, dayEnd, [] ) }
-					requestRestore={ this.handleRequestRestore }
-					siteId={ siteId }
-					tsEndOfSiteDay={ dayEnd }
-				/>
-			);
+			const dayLogs = get( logsGroupedByDay, dayEnd, [] );
+
+			if ( '' === thatEmptyDay && isEmpty( dayLogs ) ) {
+				thatEmptyDay = dayEnd;
+			} else if ( '' !== thatEmptyDay && ! isEmpty( dayLogs ) ) {
+				const thisEmptyDay = m
+					.endOf( 'day' )
+					.add( 1, 'day' )
+					.valueOf();
+				activityDays.push(
+					<ActivityLogDay
+						applySiteOffset={ this.applySiteOffset }
+						key={ `empty-${ dayEnd }` }
+						tsEndOfSiteDay={ dayEnd }
+						emptyRangeDate={
+							thisEmptyDay !== thatEmptyDay ? (
+								`${ this.applySiteOffset( moment.utc( thisEmptyDay ) ).format( 'LL' ) }
+								 - ${ this.applySiteOffset( moment.utc( thatEmptyDay ) ).format( 'LL' ) }`
+							) : (
+								`${ this.applySiteOffset( moment.utc( thatEmptyDay ) ).format( 'LL' ) }`
+							)
+						}
+					/>
+				);
+				thatEmptyDay = '';
+			} else if ( ! isEmpty( dayLogs ) ) {
+				activityDays.push(
+					<ActivityLogDay
+						applySiteOffset={ this.applySiteOffset }
+						requestedRestoreActivityId={ requestedRestoreActivityId }
+						rewindConfirmDialog={ rewindConfirmDialog }
+						disableRestore={ disableRestore }
+						hideRestore={ ! rewindEnabledByConfig || ! isPressable }
+						isRewindActive={ isRewindActive }
+						key={ dayEnd }
+						logs={ dayLogs }
+						requestRestore={ this.handleRequestRestore }
+						siteId={ siteId }
+						tsEndOfSiteDay={ dayEnd }
+					/>
+				);
+			}
```

Comment:
what's our goal in this `if/else`?

is it to add in the date between empty and non-empty days?

Reply:
based on the mockup, we may consider a reducer function to generate this in one go without mutating the array. combine that with the spread operator in assigning props and I think we can eliminate the duplication of `<ActivityLogDay />`

but still… I wonder why we have `<ActivityLogDay />` also representing days without logs. Since those date ranges don't have any logs, why not create a component designed for that one purpose, of displaying those empty days?

Reply:
actually I was going to doodle here but if you are interested just ping me in Slack and we can pair on it

Reply:
> is it to add in the date between empty and non-empty days?

The first `if` branch saves the date if the current day doesn't have events and if there wasn't a previously saved date. The `else if` branch checks if there's a saved date and if the current day has events, which would mean we have to stop the empty day range. So it renders the empty day range. The final `else if` adds a day with events.

I did think about replacing the `<ActivityLogDay />` but that one has a lot of markup and classes leading to associated styling that we would have to duplicate otherwise and always update if the `<ActivityLogDay />` styling changes. Also, the way it composes and adds the date would have to be duplicated too if we were to make a different component.

Reply:
to that point I think if we wanted to look at splitting up this component we should consider leaving all the header, placeholder, and footer stuff in there then create something like an `<ActivityLogJournal />` or `<ActivityLogView />` or `<ActivityLogStream />` or whatever and let it do the processing and rendering of those events.